packages/twenty-server/src/engine/twenty-orm/repository/workspace.repository.ts

```diff
@@ -1,7 +1,593 @@
+    const computedConditions = this.transformOptions({ where: conditions });
+
+    return this.increment(computedConditions.where, propertyPath, value);
+  }
+
+  override decrement(
+    conditions: FindOptionsWhere<Entity>,
+    propertyPath: string,
+    value: number | string,
+  ): Promise<UpdateResult> {
+    const computedConditions = this.transformOptions({ where: conditions });
+
+    return this.decrement(computedConditions.where, propertyPath, value);
+  }
+
+  /**
+   * PRIVATE METHODS
+   */
+  private getCompositeFieldMetadataArgs() {
+    const objectLiteral = ObjectLiteralStorage.getObjectLiteral(
+      this.target as any,
+    );
+
+    if (!objectLiteral) {
+      throw new Error('Object literal is missing');
+    }
+
+    const fieldMetadataArgsCollection =
+      metadataArgsStorage.filterFields(objectLiteral);
+    const compositeFieldMetadataArgsCollection =
+      fieldMetadataArgsCollection.filter((fieldMetadataArg) =>
+        isCompositeFieldMetadataType(fieldMetadataArg.type),
+      );
+
+    return compositeFieldMetadataArgsCollection;
+  }
+
+  private transformOptions<
+    T extends FindManyOptions<Entity> | FindOneOptions<Entity> | undefined,
+  >(options: T): T {
+    if (!options) {
+      return options;
+    }
+
+    const transformedOptions = { ...options };
+
+    transformedOptions.where = this.formatData(options.where);
+
+    return transformedOptions;
+  }
+
+  private formatData<T>(data: T): T {
+    if (!data) {
+      return data;
+    }
+
+    if (Array.isArray(data)) {
+      return data.map((item) => this.formatData(item)) as T;
+    }
+    const compositeFieldMetadataArgsCollection =
+      this.getCompositeFieldMetadataArgs();
+    const compositeFieldMetadataArgsMap = new Map(
+      compositeFieldMetadataArgsCollection.map((fieldMetadataArg) => [
+        fieldMetadataArg.name,
+        fieldMetadataArg,
+      ]),
+    );
+    const newData: object = {};
+
+    for (const [key, value] of Object.entries(data)) {
+      const fieldMetadataArgs = compositeFieldMetadataArgsMap.get(key);
+
+      if (!fieldMetadataArgs) {
+        if (typeof value === 'object') {
+          newData[key] = this.formatData(value);
+        } else {
+          newData[key] = value;
+        }
+        continue;
+      }
+
+      const compositeType = compositeTypeDefintions.get(fieldMetadataArgs.type);
+
+      if (!compositeType) {
+        continue;
+      }
+
+      for (const compositeProperty of compositeType.properties) {
+        const compositeKey = computeCompositeColumnName(
+          fieldMetadataArgs.name,
+          compositeProperty,
+        );
+        const value = data?.[key]?.[compositeProperty.name];
+
+        if (value === undefined || value === null) {
+          continue;
+        }
+
+        newData[compositeKey] = data[key][compositeProperty.name];
+      }
+    }
+
+    return newData as T;
+  }
+
+  private formatResult<T>(data: T): T {
+    if (!data) {
+      return data;
```

Comment:
`return;`

Reply:
Hmm I don't think so, as `null` can be a special case instead of `undefined`

Reply:
I agree with you but can it be undefined? Does the caller care about it being undefined or null? If yes, I would explicitly return `null` here.
Anyway, not really important!